src/Compilers/Core/Portable/SourceGeneration/GeneratorAttribute.cs

```diff
@@ -11,6 +11,43 @@ namespace Microsoft.CodeAnalysis
     [AttributeUsage(AttributeTargets.Class)]
     public sealed class GeneratorAttribute : Attribute
     {
-        // https://github.com/dotnet/roslyn/issues/: we don't know if we'll keep this, but for now it lets us re-use the analyzer discovery mechanism
+        /// <summary>
```

Comment:
❓ This is the same layout as the `DiagnosticAnalyzer` attribute, so that we can re-use the lookup logic. Should we enforce this via some stronger mechanism like inheritance (does that break binary compat?) or are we confident with tests covering it working as expected?

Reply:
I'm fine with it being separate. I think inheritance may be over doing it a bit here.